app/models/response.rb

```diff
@@ -315,6 +321,16 @@ def check_in!
     self.save!
   end
 
+  def generate_shortcode
+    begin
+      response_code = CODE_LENGTH.times.map { CODE_CHARS.sample }.join
+      mission_code = mission.shortcode
+      form_code = form.code
```

Comment:
This makes me kind of nervous since it's the only possible `nil` here. Didn't we discuss putting like `000` or `xxx` or something if it's nil, just to avoid a fatal error? If so, perhaps put a comment that this shouldn't happen and why, but JIC?